trio/_core/_tests/test_asyncgen.py

```diff
@@ -60,9 +64,12 @@ async def async_main():
         gc_collect_harder()
 
         # No problems saving the geniter when using either of these patterns
-        async with contextlib.aclosing(example("exhausted 3")) as aiter:
-            saved.append(aiter)
+        aiter = example("exhausted 3")
+        saved.append(aiter)
```

Comment:
`saved.append()` should be inside the `try:` to preserve original semantics, though in practice that line's unlikely to raise exceptions.

Reply:
Changed in c06e788